### ISSUE TITLE
Inconsistent undefined handling between original props and REQUEST_PRE_SEND overrides

### ISSUE THREAD

**Issue by CezarCretu** [rpldy/react-uploady#263]
**Describe the bug**
Passing a parameter whose value is `undefined` in the initial `Uploady` props creates a request that does not contain that parameter. Setting that same parameter to `undefined` by using `REQUEST_PRE_SEND` produces a request that contains the parameter with a value of `undefined`.

**To Reproduce**
Given `const foo = undefined;`,

```
<Uploady
      destination={{
        filesParamName: 'file',
        url: '...',
        params: {
           foo
        }
      }}
    />
```

Uploady produces this form data:
```
------WebKitFormBoundary
Content-Disposition: form-data; name="file"; filename="..."
Content-Type: text/xml
```

However,

```
<Uploady
      destination={{
        filesParamName: 'file',
        url: '...',
        params: {
           foo
        }
       listeners: {
         [UPLOADER_EVENTS.REQUEST_PRE_SEND]: ({items, options}) => ({options: { params: { foo } }})
       }
      }}
    />
```

produces the following:

```
------WebKitFormBoundary
Content-Disposition: form-data; name="foo"

undefined
------WebKitFormBoundary
Content-Disposition: form-data; name="file"; filename="..."
Content-Type: text/xml
```

**Expected behavior**
I expect the two requests to be identical.

**Versions**
@rpldy/uploady 0.14.2
Chrome 95

**Comment by yoavniran** [comment]
thanks @CezarCretu 
is it just the (lack of) consistency that you find problematic? Or is there one scenario you prefer? 
I actually didnt think FormData will send undefined values. I thought in the past it ignored these params. 
In any case, my inclination is to remove `undefined` params from the data being sent. However, im just wondering if there is anyone that relies on this behavior. So I might add a prop to control this

**Comment by CezarCretu** [comment]
The lack of consistency is definitely the main issue. Regardless of which behavior you end up choosing, I think both extension points should have the same behavior.

As for which one is best, I think it's unequivocal that not including `undefined` values at all is better than adding those keys with a value of `undefined`. This distinction between `undefined` and something being not defined is very JS-specific in the first place (and maybe your server isn't running JS), but in this specific scenario I don't know how your back-end would distinguish between `undefined` as a value type and the string `'undefined'`.

That being said, after looking at it more I realize the browser's native `FormData` will actually serialize nulls to `null` and undefined to `undefined`, and there is something to be said about conforming to the ecosystem's choices.

Looks like the issue I raised stems from the different merging strategies used for the [initial settings object](https://github.com/rpldy/react-uploady/blob/987966bba094cc237b10395b5623d8848eb73780/packages/core/uploader/src/uploader.js#L46) and the overrides [returned by REQUEST_PRE_SEND](https://github.com/rpldy/react-uploady/blob/e383d089987403c8c9e1417fa207979daa9ca2cb/packages/core/uploader/src/queue/processBatchItems.js#L43).

**Comment by yoavniran** [comment]
@CezarCretu merged PR #264 that will make the default behavior ignore undefined whether set from initial or preSend event/hook. Will publish 0.15.0 with the changes soon
